ClinGen/clincoded
# Connects to #447. Connects to #444. Connects to #449. Connects to #476. kl redesign summary page

This branch has solved issues in #444, #447, #449 and #476. Changes included:
* Add Not Assessment in assessment count.
* Design a new summary page to display and explain summary data and classification in clinical and experimental matrixes.
* Generate a new logic to count assessments and calculate summary to avoid mistake and bug reported in #444 and #449.
* Remove Generate New Summary button and Edit Classification link on provisional page.

Last dev merged 0836e3c.

**Test Steps and Expect Results**

**1 Test Assessment Counts and Summary Counts and Scores in Matraxies** (#447, #449 and #476)
* Create GDM: SMAD3-ORPHA284984-Autosomal unknown.
* Add pmid 123.
* Add 4 proband individuals, name them ind001, ind002, ind003 and ind004 and associate variants 111, 222, 333 and 444 respectively.
* Assess variant 111 to Supports, variant 222 to Review and variant 333 to Contradicts. Generate a summary.
Expect Result:
![summary](https://cloud.githubusercontent.com/assets/9206696/11315532/4a7598d8-8fa8-11e5-831a-0f1b8bd3abd3.png)
![summary](https://cloud.githubusercontent.com/assets/9206696/11315539/6f6b51a0-8fa8-11e5-9295-9cd66928aead.png)
* Note: Generate New Summary button is no longer displayed in provisional page.
* Click Cancel to go back to curation page.

* Add 2 families, family010 and family020. Enter segregation data each and assess family010 Supports and family020 contradicts. Click Generate Summary button.
Expect Result:
![summary](https://cloud.githubusercontent.com/assets/9206696/11315664/a7102c8c-8faa-11e5-9777-4d2ce461c092.png)
*Click Cancel

* Add 2 experimental evidence, 1 Biochemical Function named expAAA and assessed Supports and 1 Functional Alternation named expBBB and assessed Review.
* Click Generate New Summary, expect result:
Expect Assessment counts and summary scores as:
![summary](https://cloud.githubusercontent.com/assets/9206696/11375808/36e1da88-9292-11e5-84ac-dcb63e6ae428.png)
* Click Save
* Log out

**2 Test Summary Counts and Scores generated by Multiple Users (#444)**

* Log in as different user.
* Go to the same GDM curation page.
* Add article 234.
* Add a proband individual ind005, associate to variant 444.
* Assess all variants (111, 222, 333 and 444) to Support.
* Add 2 new experimental evidence, 1 Model Systems (Animal Model) nemed expCCC and 1 Rescue (Patient cells) named expDDD. Assess both Support.
* Assess expAAA and expBBB in PMID 123 to Support.
* Click Generate New Summary.
Expect to see summary counts and scores as:
![summary](https://cloud.githubusercontent.com/assets/9206696/11376328/e9b2871e-9294-11e5-9a7a-5592949d59d5.png)
* Click Save and logout.

* Login as the 3rd user.
* Go to the GDM page.
* Add article 345.
* Add a non-proband individual ind000, associate variant 111 to it.
* Add a new experimental Rescure (Engineered Equivalent) name it expXXX, associate variant 999 to it and assess the experimental as Support.
* Assess all variants (111, 222, 333, 444 and 999) to Support.
* Assess experimental evidence (expAAA, expBBB in PMID 123 and expCCC, expDDD in PMID 234) to Support.
* Generate a new summary.
Expect to see counts and scores as:
![summary](https://cloud.githubusercontent.com/assets/9206696/11376719/2603304a-9297-11e5-8adb-fd0f50a88efc.png)
* Please note that only assessment counts change. Summary scores should not change because of no new proband added. 
* Click Cancel
* In PMID 345, add proband individual ind006, associate variant 222 to it. Add experimental expEEE (Functional Alteration Patient cells), assess it as Support.
* In PMID 123, assess Segregation in family010 Contradicts and Segregation in family020 to Review. Generate a new summary.
Expect to see:
![summary](https://cloud.githubusercontent.com/assets/9206696/11376923/35179322-9298-11e5-8c36-6dd7860cd289.png)
* Save it and Logout.

* Login as the 1st user.
* Go to the GDM page.
* Add 2 new articles PMID 456 and PMID 567.
* In PMID 456, add 2 proband individuals ind007 and ind008, associate variant 222 to them. In PMID 567, add 2 proband individuals ind009 and ind010 and associate variant 111 to them.
* Change assessment values of variants 222, 333 and 444 to Support.
* Assess all experimental evidence in PMID 234 and 345 to Support.
* In PMID 123, change assessment value of expBBB to Support.
* Generate a new summary. Expect to see:
![summary](https://cloud.githubusercontent.com/assets/9206696/11575192/9b432818-99c4-11e5-849b-bf378010266d.png)
* Save it.

**End of Test**
